code/script.js

```diff
@@ -247,53 +250,86 @@ const generateBoard = () => {
-  // This variable stores what option group (category) the question belongs to.
-  // We also need a variable that stores the actual value of the question we've selected.
-
+  //console.log("categoty", category)
+  value = questions.value
+  //console.log("value", value)
+  // currentQuestion stores what attribute we ask for, the value of the attribute and what category it belongs to of the question selected
   if (category === 'hair color') {
     currentQuestion = {
       attribute: 'hairColor',
-      // value: ,
-      // 👆 add the value from the input here
-      category: category,
+      value,
+      category,
     }
   } else if (category === 'eye color') {
-    // Set this up your self
+    currentQuestion = {
+      attribute: 'eyeColor',
+      value,
+      category,
+    }
+    
   } else if (category === 'accessories') {
     currentQuestion = {
-      //attribute: ,
-      // 👆 this is the property of the booleans such as smoke, glasses and hat. add the value from the input here
-      value: true, // we're asking if this person wears a hat for exaple, so always true in the question.
-      category: category,
+      attribute: value, // this is the property of the booleans such as glasses and hat.
+      value: true, // asking if this person wears a hat for exaple, so always true in the question.
+      category,
     }
   } else if (category === 'other') {
-    // Set this up your self (should be same structure as above)
+    currentQuestion = {
+      attribute: value, 
+      value: true,
+      category,
+    }
   }
 }
 
-// This function should be invoked when you click on 'Find Out'.
+// This function is invoked when you click 'Find Out'.
+// Compare the currentQuestion with the secret person. Keep or remove characters based on that
 const checkQuestion = () => {
-  // Compare the currentQuestion with the secret person.
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
+  let keep 
+  if (currentQuestion.attribute === 'hairColor' && currentQuestion.value === secret.hairColor) {
+    keep = true
+  } else if (currentQuestion.attribute === 'eyeColor' && currentQuestion.value === secret.eyeColor) {
+    keep = true
+  } else if (currentQuestion.attribute === 'glasses' && currentQuestion.value === secret.glasses) {
+    keep = true
+  } else if (currentQuestion.attribute === 'hat' && currentQuestion.value === secret.hat) {
+    keep = true
+  } else if (currentQuestion.attribute === 'smoker' && currentQuestion.value === secret.smoker) {
+    keep = true
+  } else {
+    keep = false
+  }
+  console.log("keep", keep);
+  // Invoke filterCharacters
+  filterCharacters(keep)
 }
 
-// It'll filter the characters array and redraw the game board.
+// Filter the characters array and redraw the game board.
 const filterCharacters = (keep) => {
+  let category = currentQuestion.category
+  let attribute = currentQuestion.attribute
+  let value =currentQuestion.value
```

Comment:
`const { category, attribute, value } = currentQuestion` would do the magic here 🧙🏻